### PR TITLE
Add strict mode to validate command

### PR DESCRIPTION
**Pull request by reinfi**
Resolving issue (https://github.com/cheeZery/depdoc/issues/20).

Add --strict and --very-strict option to validate command.

Used foreign library to check for version.

### CODE REVIEW

**Comment by reinfi** [comment]
Will add more tests when the other PR is resolved. Because this is kind of awkward to add tests for the commands.

**Comment by reinfi** [comment]
@GodLesZ any opinion on this?

File: README.md [hunk 1]
```diff
@@ -19,3 +19,15 @@ By adding a lock emoji (🔒) after the version number, you can document that th
 ```
 
 Validates that every installed dependency is documented in DEPENDENCIES.md. Also makes sure that no package surpasses its locked version.
+
+Use strict modes to verify your package version matches the documented dependencies.
+
+```
+./vendor/bin/depdoc validate --strict
+```
+Checks for major and minor version to match, i.e. installed 1.0.1 and documented 1.0.0 will bypass validation.
```
Comment (GodLesZ) [inline README.md]:
```suggestion
Checks for major and minor version to match, i.e. installed **1.0.1** and documented **1.0.0** will bypass validation but installed **1.1.0** and documented **1.0.0** won't.
```

Just a small gain to clarify

File: src/Validator/StrictMode.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    protected const MAJOR_AND_MINOR = 1;
+    protected const FULL_SEM_VER_MATCH = 2;
+
+    /** @var int */
+    private $mode;
+
+    /**
+     * @param int $mode
+     */
+    protected function __construct(int $mode)
```
Comment (GodLesZ) [inline src/Validator/StrictMode.php]: Why ever this must be an factory/builder pattern implementation :D
But it won't hurt so ¯\_(ツ)_/¯

**Comment by GodLesZ** [comment]
> @GodLesZ any opinion on this?

This looks awesome!
Sorry for the late reply

**Pull request merged**